src/sse_events.rs

```diff
@@ -0,0 +1,154 @@
+        match cloned_co {
+            ConsoleOutput::Chat(mut m) => {
+                let players = state.players.get_name_to_steam_ids_map();
+                if let Some(id) = players.get(&m.player_name) {
+                    m.set_steam_id(*id);
+                }
+                let event = SerializableEvent::make_from(m);
+                Some(serde_json::to_string(&event).expect("Serialisation failure"))
+            }
+            ConsoleOutput::Kill(mut m) => {
+                let players = state.players.get_name_to_steam_ids_map();
+                if let Some(id) = players.get(&m.killer_name) {
+                    m.set_steam_id_killer(*id);
+                }
+                if let Some(id) = players.get(&m.victim_name) {
+                    m.set_steam_id_victim(*id);
+                }
+                let event = SerializableEvent::make_from(m);
+                Some(serde_json::to_string(&event).expect("Serialisation failure"))
+            }
+            ConsoleOutput::DemoStop(m) => {
+                let event = SerializableEvent::make_from(m);
+                Some(serde_json::to_string(&event).expect("Serialisation failure"))
+            }
+            _ => None,
+        }
+    }
+
+    /// Handling `DemoMessages` often requires inspecting or modifying the list of votes that have been
+    /// cast. I.e. a `VoteCast` event wont contain information about what the vote options were. So we
+    /// have to keep this context ourselves.
+    fn handle_demo_message(&mut self, state: &MACState, message: &DemoMessage) -> Option<String> {
+        let cloned_msg = message.clone();
+        match cloned_msg.event {
+            DemoEvent::VoteStarted(options) => {
+                let mut values = Vec::new();
+                for i in 0..options.count {
+                    let opt = match i {
+                        0 => options.option_1.to_string(),
+                        1 => options.option_2.to_string(),
+                        2 => options.option_3.to_string(),
+                        3 => options.option_4.to_string(),
+                        4 => options.option_5.to_string(),
+                        _ => String::new(),
+                    };
+                    values.push(opt);
+                }
+                self.votes.insert(options.voteidx, values);
+
+                let res = VoteRelatedEvent::make_from(options);
+                Some(serde_json::to_string(&res).expect("Serialisation failure"))
+            }
+            DemoEvent::VoteCast(event, steamid) => {
+                let name = steamid
+                    .as_ref()
+                    .and_then(|&id| state.players.get_name(id))
+                    .unwrap_or("Someone");
+
+                let vote: &str = self
+                    .votes
+                    .get(&event.voteidx)
+                    .and_then(|v| v.get(event.vote_option as usize))
+                    .map_or::<&str, _>("Invalid vote", |s| s);
```

Comment:
I know this was just copied from the `PrintVotes` handler, but I think for the SSE messages it would probably be a good idea to provide better defaults, I noticed the struct you use has `Option`s, so maybe they should just be `None` in the fail case.

Reply:
+1